tests/tracer/test_encoders.py

```diff
@@ -391,12 +258,36 @@ class SubFloat(float):
 )
 def test_span_types(span, tags):
     refencoder = RefMsgpackEncoder()
-    encoder = MsgpackEncoder()
+    encoder = MsgpackEncoder(1 << 10, 1 << 10)
 
     span.set_tags(tags)
 
     # Finish the span to ensure a duration exists.
     span.finish()
 
     trace = [span]
-    assert decode(refencoder.encode_trace(trace)) == decode(encoder.encode_trace(trace))
+    encoder.put(trace)
+    assert decode(refencoder.encode_traces([trace])) == decode(encoder.encode())
+
+
+@given(
+    name=text(),
+    service=text(),
+    resource=text(),
+    meta=dictionaries(text(), text()),
+    metrics=dictionaries(text(), floats()),
+    error=integers(),
+    span_type=text(),
+)
+@settings(max_examples=200)
+def test_custom_msgpack_encode_trace_size(name, service, resource, meta, metrics, error, span_type):
+    encoder = MsgpackEncoder(1 << 20, 1 << 20)
+    span = Span(tracer=None, name=name, service=service, resource=resource)
+    span.meta = meta
+    span.metrics = metrics
+    span.error = error
+    span.span_type = span_type
+    trace = [span, span, span]
+
+    encoder.put(trace)
+    assert encoder.size + 1 == len(encoder.encode())
```

Comment:
Why is the encoder's expected size to be the size of the encoded trace + 1?

Reply:
`encoder.encode()` adds an extra byte at the front, which represents the size of the resulting array, whereas `encoder.size` only accounts for the actual data that's been encoded so far.